dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -178,6 +178,39 @@ public static CloseableClient createCloseableHttpClient(ClientConfiguration conf
         return new CloseableClient(builder.build());
     }
 
+    private static void setupConnectionPoolMetrics(
+            TaggedMetricRegistry taggedMetrics,
+            String channelName,
+            PoolingHttpClientConnectionManager connectionManager) {
+        WeakSummingGauge.getOrCreate(
+                pool -> pool.getTotalStats().getAvailable(),
+                connectionManager,
+                taggedMetrics,
+                MetricName.builder()
+                        .safeName("dialogue.client.pool.size")
```

Comment:
shame we can't use metric-schema here, this is one of the most important sets of metrics (for me anyhow, service authors shouldn't need to worry about these unless something has gone south)

Reply:
I think if metric schema exposed just the MetricName we could use it?